resetcloser_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+type testCloser struct {
+	closed bool
+}
+
+func (tc *testCloser) Close() error {
+	tc.closed = true
+	return nil
+}
+
+func TestResetCloser(t *testing.T) {
+	var tc testCloser
+	assert.False(t, tc.closed)
+	func() {
+		mc := NewResetCloser(&tc)
+		defer mc.Close()
+	}()
+	assert.True(t, tc.closed)
+}
+
+func TestResetCloserReset(t *testing.T) {
+	var tc testCloser
+	assert.False(t, tc.closed)
+	func() {
+		mc := NewResetCloser(&tc)
+		mc.Reset()
+		defer mc.Close()
+	}()
+	assert.False(t, tc.closed)
+}
+
+func TestTwoResetClosers(t *testing.T) {
+	var tc1, tc2 testCloser
+	assert.False(t, tc1.closed)
+	assert.False(t, tc2.closed)
+	func() {
+		mc := NewResetCloser(&tc1)
+		defer mc.Close()
+		mc = NewResetCloser(&tc2)
```

Comment:
Actually this test was written so that I could confirm that the first defer would continue to hold a reference to the original `ResetCloser`, even after it gets overwritten. Maybe this was more useful for my own understanding though, rather than to verify behaviour, so maybe it doesn't need to be here...

Reply:
I've deleted this test.